ui/src/pages/network/Trading.tsx

```diff
@@ -15,39 +18,44 @@ export const TradingServiceTable: React.FC<Props> = ({ services }) => {
   const { getName } = usePartyName(party);
   const ledger = useLedger();
 
+  const role = useStreamQueries(ExchangeRole).contracts.find(rl => rl.payload.provider === party);
```

Comment:
idea for future PR - `useRole` hooks

Reply:
yeah that would be great! I just tried adding it to `RolesContext` quickly and I was getting type errors (possibly because mixing Roles and Services?) so I'll leave it like this for now